GDAS-validation/setup_workspace.sh

```diff
@@ -0,0 +1,154 @@
+elif [ $machine = hera ]; then
+  workdir=/scratch1/NCEPDEV/stmp2/$LOGNAME/gdas-validation
+  ICSDir=/scratch1/NCEPDEV/da/Cory.R.Martin/blah/blah
+else
+   echo "Machine " $machine "not found"
+   exit 1
+fi
+
+EXPNAME="gdas_eval_satwind"
+mydir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+
+#-------------- User should not modify below here ----------
+mkdir -p $workdir
+
+#--- clone repositories ---
+if [ $clone = "YES" ]; then
+  cd $workdir
+  echo "Cloning global-workflow at $workdir/global-workflow"
+  git clone https://github.com/noaa-emc/global-workflow.git
+  cd global-workflow/sorc
+  ./checkout.sh -g
+  rm -rf gsi_enkf.fd
+  git clone --recursive https://github.com/NOAA-EMC/GSI.git gsi_enkf.fd
+  git clone --recursive https://github.com/NOAA-EMC/GDASApp.git gdas.cd
+fi
+
+#--- build GDASApp and GSI ---
+if [ $build = "YES" ]; then
+  cd $workdir/global-workflow/sorc/gsi_enkf.fd/ush
+  echo "Building GSI in $workdir/global-workflow/sorc/gsi_enkf.fd/"
+  echo "Build begin: `date`"
+  echo "Build log: $workdir/build_gsi.log"
+  ./build.sh > $workdir/build_gsi.log 2>&1
+  echo "Build complete: `date`"
+  cd $workdir/global-workflow/sorc/gdas.cd
+  echo "Building GDASApp in $workdir/global-workflow/sorc/gdas.cd"
+  echo "Build begin: `date`"
+  echo "Build log: $workdir/build_gdasapp.log"
+  WORKFLOW_BUILD="ON" ./build.sh > $workdir/build_gdasapp.log 2>&1
+  echo "Build complete: `date`"
+  cd $workdir/global-workflow/sorc/
+  echo "Link workflow"
+  ./link_workflow.sh
+  # copy workflow default config files
+  echo "Staging configuration files"
+  mkdir -p $workdir/gdas_config
+  cp -rf $workdir/global-workflow/parm/config/gfs/* $workdir/gdas_config/.
+  # copy files that need to be overwritted from default
+  cp -rf $mydir/gdas_config/* $workdir/gdas_config/.
+fi
+
+#--- setup default experiment within workflow
+if [ $setup = "YES" ]; then
+  module use $workdir/global-workflow/sorc/gdas.cd/modulefiles
+  module load GDAS/$machine
+  cd $workdir/global-workflow/workflow
+  # setup_expt variables
+  IDATE=2021080100
+  EDATE=2021080200
+  RESDET=768
+  CDUMP=gdas
+  PSLOT=${EXPNAME:-"gdas_eval"}
+  CONFIGDIR=$workdir/gdas_config
+  COMROT=$workdir/comrot
+  EXPDIR=$workdir/expdir
+  ICSDIR=$ICSDir/$IDATE
+  rm -rf $EXPDIR/${PSLOT}*
+  rm -rf $COMROT/${PSLOT}*
+  # make two experiments, one GSI, one JEDI
+  set -x
+  ./setup_expt.py gfs cycled --idate $IDATE --edate $EDATE --app ATM --start warm --gfs_cyc 0 \
+    --resdet $RESDET  --nens 0 --cdump $CDUMP --pslot ${PSLOT}_GSI --configdir $CONFIGDIR \
+    --comrot $COMROT --expdir $EXPDIR --yaml $CONFIGDIR/config_gsi.yaml --icsdir $ICSDIR
+  ./setup_expt.py gfs cycled --idate $IDATE --edate $EDATE --app ATM --start warm --gfs_cyc 0 \
+    --resdet $RESDET  --nens 0 --cdump $CDUMP --pslot ${PSLOT}_JEDI --configdir $CONFIGDIR \
+    --comrot $COMROT --expdir $EXPDIR --yaml $CONFIGDIR/config_jedi.yaml --icsdir $ICSDIR
+  set +x
+  # setup the two XMLs
```

Comment:
Minor comment:  We could execute `setup_xml.py` immediately after  `setup_expt.py` for each experiment.   With this refactoring we see the following written to stdout
```
./setup_expt.py for GSI experiment
EDITED: GSI config.base
Finalizing initialize
...
 ./setup_expt.py for JEDI experiment
EDITED: JEDI config.base
Finalizing initialize
...
Linking backgrounds ...
```
The GSI and JEDI setup are separate and distinct blocks in stdout.   

Currently we have

```
./setup_expt.py for GSI experiment
EDITED: GSI config.base
 ./setup_expt.py for JEDI experiment
EDITED: JEDI config.base
Finalizing initialize
...
Finalizing initialize
...
Linking backgrounds ...
```